### PR TITLE
chore(gatsby): migrate query/redirectWriter to TypeScript

### PR DESCRIPTION
**Pull request by taejs**
## Description
Migrate query/redirectWriter to TypeScript

## Related Issues
Related to #21995

### CODE REVIEW

File: packages/gatsby/src/query/redirects-writer.ts [hunk 1]
```diff
@@ -31,18 +34,14 @@ const writeRedirects = async () => {
...
-let oldRedirects
 const debouncedWriteRedirects = _.debounce(() => {
   // Don't write redirects again until bootstrap has finished.
   if (
     bootstrapFinished &&
     !_.isEqual(oldRedirects, store.getState().redirects)
   ) {
     writeRedirects()
-    oldRedirects = store.getState().Redirects
+    oldRedirects = store.getState().redirects
```
Comment (taejs) [inline packages/gatsby/src/query/redirects-writer.ts:44]: I correct this code because I think it's a typo

**Comment by pieh** [comment]
Hey @taejs! 

This seems to be duplicate of https://github.com/gatsbyjs/gatsby/pull/23332. I'm just looking, at that was not tracked in typescript migration umbrella issue - so sorry about that. Given that you are already member of gatsby org because of previous pull requests I would like to give a chance to other PR as this is author's first if that's fine with you?

File: packages/gatsby/src/query/redirects-writer.ts (hunk 1 above)
Comment (pieh) [reply packages/gatsby/src/query/redirects-writer.ts:44]: That's nice catch!

But there is some asterisk regarding this entire file - we actually don't use it anywhere 😱 

It seems after we moved this file to https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/bootstrap/redirects-writer.js there was some git messup and old file location stayed. What's really weird is that the file we actually use - don't even have that check

**Comment by taejs** [comment]
Sure!  no problem

**Pull request closed without merging**